decorator.js

```diff
@@ -21,11 +21,11 @@ function addClickOutsideListener(Component) {
     },
 
     render: function render() {
-      return React.createElement(Component, objectAssign({
+      return React.createElement(Component, objectAssign({}, this.props, {
```

Comment:
I may also be asking it wrong - the fact that the assignment order matters indicates that we two things that do completely different things with the same name, so that feels like a thing that needs to be fixed by chosing new names and releasing a new major version that breaks backward compatibility. And probably rewriting the HOC to then be a straight-up wrapper (with a props passthrough) instead of doing what it does right now.